sycl/doc/design/DeviceAspectTraitDesign.md

```diff
@@ -0,0 +1,128 @@
+# Implementation design for `sycl::any_device_has` and `sycl::all_devices_have`
+
+This design document describes the implementation of the SYCL 2020 device aspect
+traits `any_device_has` and `all_devices_have` as described in the
+[SYCL 2020 Specification Rev. 6 Section 4.6.4.3][1].
+
+In summary, `any_device_has<aspect>` and `all_devices_have<aspect>` must inherit
+from either `std::true_t` or `std::false_t` depending on whether the
+corresponding compilation environment can guarantee that any and all the
+supported devices support the `aspect`.
+
+The design of these traits is inspired by the implementation of the
+[sycl\_ext\_oneapi\_device\_if][2] and
+[sycl\_ext\_oneapi\_device\_architecture][3] extensions as described in
+[DeviceIf.md][4]. Additionally, it leverages part of the design for optional
+kernel features, as described in [OptionalDeviceFeatures.md][5].
+
+## Changes to the compiler driver
+
+Using the `-fsycl-targets` options introduced in [DeviceIf.md][4] and the
+configuration file introduced in [OptionalDeviceFeatures.md][5], the compiler
+driver finds the set of all aspects supported by each specified target. Note
+that in this section we refer to aspects as their integral representation as
+specified in the device headers rather than by the names specified in the
+[SYCL 2020 specification][1].
+
+For each target $t$ in `-fsycl-targets`, let $A^{any}_t$ be the set of aspects
+supported by any device supporting $t$ and let $A^{all}_t$ be the set of aspects
+supported by all devices supporting $t$. These sets are defined as follows:
+* If $t$ has an entry in the configuration file, $A^{all}_t$ is the same as the
+`aspects` list in that entry.
+* If $t$ does not have an entry in the configuration file, $A^{all}_t$ is empty.
+* If $t$ has an entry in the configuration file and the entry has a value
```

Comment:
Do I read it correctly that we intend this `may_support_other_aspects` value to be optional? In any case, I suggest that we expand "Device configuration file" section of optional kernel features document to explicitly mention the value, because it only exists there in the example

Reply:
Something like 114a8340b7cdb4c0baf63c84629a147ef6cded0b?

Reply:
Yes, thanks!